### PR TITLE
docs: update development guide with proper installation steps.

### PR DESCRIPTION
**Pull request by qkaiser**
Lack of proper identification was identified when @DenysVznk set up their environment.

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -90,8 +90,34 @@ pyenv local 3.8.12
...
 ### Installing Python dependencies
 
-We are using [poetry](https://python-poetry.org/) to manage our Python dependencies. `poetry
-install` will install all required dependencies for development.
+We are using [poetry](https://python-poetry.org/) to manage our Python
+dependencies. To install all required dependencies for development, you can run
+the following command:
+
+```
+poetry install --with dev,docs
```
Comment (vlaci) [inline docs/development.md:98]: do we need docs by default?

Comment (qkaiser) [reply docs/development.md]: no :)

Comment (qkaiser) [reply docs/development.md]: removed it

File: docs/development.md [hunk 2]
```diff
@@ -90,8 +90,34 @@ pyenv local 3.8.12
...
+poetry run unblob
+poetry run pytest tests -v
+```
+
+By dropping into the virtual environment:
+
+```
+$ poetry shell
+Spawning shell within /home/user/.cache/pypoetry/virtualenvs/unblob-FWlmBZih-py3.11
+$ emulate bash -c '. /home/quentin/.cache/pypoetry/virtualenvs/unblob-FWlmBZih-py3.11/bin/activate'
```
Comment (vlaci) [inline docs/development.md:117]: Ths seems a  `zshism` to me. Also activate shouldn't be needed if already in poetry shell.

Comment (qkaiser) [reply docs/development.md]: just kept the commands, removed everything else

**Pull request merged**